src/Handler/Table/Preview/PreviewTableHandler.php

```diff
@@ -0,0 +1,166 @@
+use Google\Protobuf\NullValue;
+use Google\Protobuf\Value;
+use Keboola\Datatype\Definition\BaseType;
+use Keboola\Datatype\Definition\Teradata;
+use Keboola\StorageDriver\Command\Table\PreviewTableCommand;
+use Keboola\StorageDriver\Command\Table\PreviewTableResponse;
+use Keboola\StorageDriver\Contract\Driver\Command\DriverCommandHandlerInterface;
+use Keboola\StorageDriver\Credentials\GenericBackendCredentials;
+use Keboola\StorageDriver\Teradata\TeradataSessionManager;
+use Keboola\TableBackendUtils\Column\Teradata\TeradataColumn;
+use Keboola\TableBackendUtils\Escaping\Teradata\TeradataQuote;
+use Keboola\TableBackendUtils\Table\Teradata\TeradataTableReflection;
+
+class PreviewTableHandler implements DriverCommandHandlerInterface
+{
+    public const STRING_MAX_LENGTH = 50;
+
+    private TeradataSessionManager $manager;
+
+    public function __construct(TeradataSessionManager $manager)
+    {
+        $this->manager = $manager;
+    }
+
+    /**
+     * @inheritDoc
+     * @param GenericBackendCredentials $credentials
+     * @param PreviewTableCommand $command
+     */
+    public function __invoke(
+        Message $credentials,
+        Message $command,
+        array $features
+    ): ?Message {
+        assert($credentials instanceof GenericBackendCredentials);
+        assert($command instanceof PreviewTableCommand);
+
+        // validate
+        assert($command->getPath()->count() === 1, 'CreateTableCommand.path is required and size must equal 1');
+        assert(!empty($command->getTableName()), 'CreateTableCommand.tableName is required');
+        assert($command->getColumns()->count() > 0, 'CreateTableCommand.columns is required');
+
+        try {
+            $db = $this->manager->createSession($credentials);
+            /** @var string $databaseName */
+            $databaseName = $command->getPath()[0];
+
+            // build sql
+            $columns = $this->repeatedStringToArray($command->getColumns());
+            // TODO changeSince, changeUntil
+            // TODO fulltextSearch
+            // TODO whereFilters
+            $selectTableSql = sprintf(
+                "SELECT %s %s\nFROM %s.%s",
+                $command->getLimit() ? sprintf('TOP %d', $command->getLimit()) : '',
+                implode(', ', array_map([TeradataQuote::class, 'quoteSingleIdentifier'], $columns)),
+                TeradataQuote::quoteSingleIdentifier($databaseName),
+                TeradataQuote::quoteSingleIdentifier($command->getTableName())
+            );
+
+            if ($command->hasOrderBy() && $command->getOrderBy()) {
+                $orderBy = $command->getOrderBy();
+                $selectTableSql .= sprintf(
+                    "\nORDER BY %s %s",
+                    // TODO k cemu je potreba dataType?
+                    TeradataQuote::quoteSingleIdentifier($orderBy->getColumnName()),
+                    $orderBy->getOrder() === PreviewTableCommand\PreviewTableOrderBy\Order::ASC ? 'ASC' : 'DESC'
```

Comment:
TODO k cemu je potreba `dataType` u `order by`?

Reply:
castuje se to tam nejak https://github.com/keboola/connection/blob/28b98d4255596eecea0198cb93b8c54c9470c02f/Package/StorageBackend/src/ImportExport/QueryBuilder/ExasolExportQueryBuilder.php#L258

Reply:
ale nevim jestli to na TD ma nejakou hodnotu

Reply:
pridal jsem castovani na `integer`, `bigint` a `real` - ty nepotrebujou definici velikosti.